### PR TITLE
add timestamp at event levetl to save_tiff

### PR DESCRIPTION
**Pull request by chiahaoliu**
also other small refactoring to make code more succinct and robust

### CODE REVIEW

File: xpdacq/analysis.py [hunk 1]
```diff
@@ -141,7 +138,7 @@ def save_tiff(headers, dark_subtraction = True, *, max_count = None):
             else:
                 print('Sorry, something went wrong with your tif saving')
                 return
-            if max_count is not None and ind<= max_count:
```
Comment (tacaswell) [inline xpdacq/analysis.py]: See, never trust me ;)

Comment (sbillinge) [inline xpdacq/analysis.py]: @chiaohaoliu if you are finished with commits/discussion and want something
pulled in, please could you make that request?  I would like to be pulling
things in in a more granular fashion than we have....for example, this
small change to ``save_tiff``, and it is easier to do that if I know that
it is passing all tests, functional and unit, and is ready to be pulled in.

S

On Mon, Mar 14, 2016 at 1:48 PM, Thomas A Caswell <notifications@github.com>
wrote:

> In xpdacq/analysis.py
> <https://github.com/xpdAcq/xpdAcq/pull/136#discussion_r56043933>:
>
> > @@ -141,7 +138,7 @@ def save_tiff(headers, dark_subtraction = True, *, max_count = None):
> >              else:
> >                  print('Sorry, something went wrong with your tif saving')
> >                  return
> > -            if max_count is not None and ind<= max_count:
>
> See, never trust me ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xpdAcq/xpdAcq/pull/136/files#r56043933>.
>



-- 
----
Prof. Simon Billinge

Applied Physics & Applied Mathematics
Columbia University
500 West 120th Street
Room 200 Mudd, MC 4701
New York, NY 10027
Tel: (212)-854-2918 (o) 851-7428 (lab)

Condensed Matter Physics and Materials Science Dept.
Brookhaven National Laboratory
P.O. Box 5000
Upton, NY 11973-5000
(631)-344-5661

email: sb2896 at columbia dot edu
home: http://thebillingegroup.com

**Comment by chiahaoliu** [comment]
Yes, I will make a statement of finishing an pull request from now on. I agree sometimes it is hard to tell the status from commit message.

**Comment by chiahaoliu** [comment]
Regarding this PR. I add functional test code that was run on real notebook server. By interacting with good amounts of real data, I found some edge cases and fixed them in last commit. I think it is ready to be pulled in, if other edits make sense for you. Thanks!

**Comment by sbillinge** [comment]
Excellent.  I will have to do it later though.

Please can you save the tests that you run as functional tests, preferably
that can execute as scripts.  We can run them again later, for example as
acceptance tests for software upgrades.  Actually we have three cases and
ideally have tests for each:
1) unit tests
2) running interactively at XPD
3) running interactively in simulation mode

On Monday, March 14, 2016, Timothy Liu <notifications@github.com> wrote:

> Regarding this PR. I add functional test code that was run on real
> notebook server. By interacting with good amounts of real data, I found
> some edge cases and fixed them in last commit. I think it is ready to be
> pulled in, if other edits make sense for you. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xpdAcq/xpdAcq/pull/136#issuecomment-196498400>.
>


-- 
Prof. Simon Billinge
Applied Physics & Applied Mathematics
Columbia University
500 West 120th Street
Room 200 Mudd, MC 4701
New York, NY 10027
Tel: (212)-854-2918 (o) 851-7428 (lab)

Condensed Matter Physics and Materials Science Dept.
Brookhaven National Laboratory
P.O. Box 5000
Upton, NY 11973-5000
(631)-344-5661

email: sb2896 at columbia dot edu
home: http:// <http://nirt.pa.msu.edu/>bgsite.apam.columbia.edu/

**Comment by chiahaoliu** [comment]
Sure, I will pack it as a executable script and push it. However, without ssh into XPD computer, I was only able to run this code by pasting it on to ipython notebook widgets on notebook server, but not executing in command line on where this notebook server is located, since environment is not configured yet (python 2.7 and dependency packages are not installed). This functional test script should be able to run at XPD for sure.

**Comment by sbillinge** [comment]
Good.  It is only needed for XPD.  We can make another script for simulation
.

S

On Monday, March 14, 2016, Timothy Liu <notifications@github.com> wrote:

> Sure, I will pack it as a executable script and push it. However, without
> ssh into XPD computer, I was only able to run this code by pasting it on to
> ipython notebook widgets on notebook server, but not executing in command
> line on where this notebook server is located, since environment is not
> configured yet (python 2.7 and dependency packages are not installed). This
> functional test script should be able to run at XPD for sure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xpdAcq/xpdAcq/pull/136#issuecomment-196528873>.
>


-- 
Prof. Simon Billinge
Applied Physics & Applied Mathematics
Columbia University
500 West 120th Street
Room 200 Mudd, MC 4701
New York, NY 10027
Tel: (212)-854-2918 (o) 851-7428 (lab)

Condensed Matter Physics and Materials Science Dept.
Brookhaven National Laboratory
P.O. Box 5000
Upton, NY 11973-5000
(631)-344-5661

email: sb2896 at columbia dot edu
home: http:// <http://nirt.pa.msu.edu/>bgsite.apam.columbia.edu/
